onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -621,12 +633,25 @@ template <>
 Status ReduceKernel<true>::ComputeImpl<int32_t, CUDNN_REDUCE_TENSOR_NO_INDICES>(OpKernelContext* ctx, cudnnReduceTensorOp_t cudnn_reduce_op) const {
   typedef typename ToCudaType<int32_t>::MappedType CudaT;
 
+  const std::string& op_name = this->KernelDef().OpName();
+  std::vector<int64_t> axes_values = axes_;
+  if (op_name == "ReduceSumTraining") {
+    //override the attribute value with the input value for reduction_axes
+    const Tensor* axesTensor = ctx->Input<Tensor>(1);
+    ORT_ENFORCE(axesTensor->Shape().NumDimensions() == 1, "An axes tensor must be a vector tensor.");
+    auto nDims = static_cast<size_t>(axesTensor->Shape()[0]);
+    auto data = onnxruntime::make_unique<int64_t[]>(nDims);
+    CUDA_RETURN_IF_ERROR(cudaMemcpyAsync(data.get(), axesTensor->template Data<int64_t>(), nDims * sizeof(int64_t), cudaMemcpyDeviceToHost));
```

Comment:
BoradcastGradientArgs should return axes tensor in CPU. 

Also enforce this in the kernel registration. 
.InputMemoryType<OrtMemTypeCPUInput>(1)  /* Keep axes in CPU */       \

so you don't need to explicitly copy it here.